[onert/docs] Write backend-api.md (#4256)
Samsung/ONE: docs/runtime/api.md

## Backend API

> We provide Backend API to allow anyone to implement so one can extend the runtime in terms of operation computation.

Reviewer: @wateret 

- What do you mean by `so` here? Shared library?
- Also, the explanation here is different from `backend-api.md`. It does not say about memory management. Which one is correct? Here or `backend-api.md`?

Author: - "so" here is an adverb, please let me know it is grammatically wrong or have a better sentence
- Both are correct. If I should summarize "Backend" as a one word, it would be "operation computation". Just wanted to keep this sentence short

I changed this part to avoid duplication, I could just remove the whole sentence 😄  or please suggest a sentence.

Reviewer: >     * "so" here is an adverb, 

Thank you for answer. 

> please let me know it is grammatically wrong or have a better sentence

I don't see grammatical error. I was just confused whether `so` is for `implement` or for following sentence.

> have a better sentence

I think you mean:

Backend API enables anyone to extend the runtime in terms of operation computation.

( I am not sure it is better sentence. I just removed the words that give less information at least to me. )

> Both are correct. If I should summarize "Backend" as a one word, it would be "operation computation".

I thought `computation` and `memory allocation` are different concept like `ComputeLibrary`'s `Core` and `Runtime`. `operation computation` is okay since details will be explained in `backend-api.md`.

Author: ```suggestion
Backend API enables anyone to extend the runtime in terms of operation computation and memory management.
```

Author: Updated the sentence, thanks!
